src/Plugin/SmsGateway/SmsIncomingEventInterface.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace Drupal\sms\Plugin\SmsGateway;
+
+use Drupal\sms\Event\SmsMessageEvent;
+
+/**
+ * Interface for gateways implementing an incoming event subscriber.
+ */
+interface SmsIncomingEventInterface {
```

Comment:
Still missed out the 'Processor' part which I think is the key work here `SmsIncomingEventProcessorInterface`